tests/page-objects/forms/generic-form.po.js

```diff
@@ -18,18 +18,27 @@ module.exports = {
     editFormBtn.click();
   },
 
+  editFormNative: async () => {
+    const editFormBtn = element(
+      by.css('[href^="#/reports/edit"]>.fa-pencil')
+    );
+    await helper.waitUntilReadyNative(editFormBtn);
+    await editFormBtn.click();
+  },
+
   goBack: () => {
     element(by.css('button.btn.btn-default.previous-page')).click();
   },
 
-  invalidateReport: () => {
-    const reportInvalidBtn = element(by.css('[ng-include*="verify-invalid"]'));
-    helper.waitUntilReady(reportInvalidBtn);
-    reportInvalidBtn.click();
-    const reportInvalidIcon = element(by.css('.detail>.status>.error'));
-    helper.waitUntilReady(reportInvalidIcon);
-    const reportInvalidMessage = element(by.css('.verify-error>span:last-of-type'));
-    expect(reportInvalidMessage.getText()).toEqual('Has errors');
+  invalidateReportNative: async () => {
+    const reportInvalidBtn = element(by.css('.verify-error'));
+    await helper.waitUntilReadyNative(reportInvalidBtn);
+    await reportInvalidBtn.click();
+    const reportInvalidIcon = element(by.css('.verify-error'));
+    await helper.waitUntilReadyNative(reportInvalidIcon);
+    await helper.waitUntilReadyNative(reportInvalidIcon);
+    const reportInvalidMessage = element(by.css('.verify-error'));
+    expect(await reportInvalidMessage.getText()).toEqual('Has errors');
```

Comment:
I'm so sorry I didn't notice this in an earlier review, I slipped by it completely. 

I see you've changed a the second icon selector here from 
```
const reportInvalidIcon = element(by.css('.detail>.status>.error'));
```
with 
```
const reportInvalidIcon = element(by.css('.verify-error'));
```

The first one is the status icon that appears in the report detail when the report is verified as invalid, and that's the one we want to test, not the one we've just clicked. 
So, this can be changed to 
```suggestion
    const reportInvalidIcon = element(by.css('.detail>.status>.error'));
    await helper.waitUntilReadyNative(reportInvalidIcon);
    const reportInvalidMessage = element(by.css('.verify-error.active'));
    expect(await reportInvalidMessage.getText()).toEqual('Has errors');
```

Reply:
Thanks, I have made the change.